client/package.json

```diff
@@ -13,51 +13,51 @@
   },
   "private": true,
   "dependencies": {
-    "@angular/animations": "^14.0.0",
-    "@angular/cdk": "^14.0.0",
-    "@angular/common": "^14.0.0",
-    "@angular/compiler": "^14.0.0",
-    "@angular/core": "^14.0.0",
-    "@angular/flex-layout": "^13.0.0-beta.38",
-    "@angular/forms": "^14.0.0",
-    "@angular/material": "^14.0.0",
-    "@angular/platform-browser": "^14.0.0",
-    "@angular/platform-browser-dynamic": "^14.0.0",
-    "@angular/router": "^14.0.0",
+    "@angular/animations": "^14.0.4",
+    "@angular/cdk": "^14.0.4",
+    "@angular/common": "^14.0.4",
+    "@angular/compiler": "^14.0.4",
+    "@angular/core": "^14.0.4",
+    "@angular/flex-layout": "^14.0.0-beta.40",
+    "@angular/forms": "^14.0.4",
+    "@angular/material": "^14.0.4",
+    "@angular/platform-browser": "^14.0.4",
+    "@angular/platform-browser-dynamic": "^14.0.4",
+    "@angular/router": "^14.0.4",
     "rxjs": "~7.5.5",
     "tslib": "^2.4.0",
-    "zone.js": "~0.11.5"
+    "zone.js": "~0.11.6"
   },
   "devDependencies": {
-    "@angular-devkit/architect": "^0.1400.0",
-    "@angular-devkit/build-angular": "^14.0.0",
-    "@angular-eslint/builder": "~13.2.1",
-    "@angular-eslint/eslint-plugin": "~13.2.1",
-    "@angular-eslint/eslint-plugin-template": "~13.2.1",
-    "@angular-eslint/schematics": "~13.2.1",
-    "@angular-eslint/template-parser": "~13.2.1",
-    "@angular/cli": "^14.0.0",
-    "@angular/compiler-cli": "^14.0.0",
-    "@angular/language-service": "^14.0.0",
-    "@cypress/schematic": "^1.7.0",
+    "@angular-devkit/architect": "^0.1400.4",
+    "@angular-devkit/build-angular": "^14.0.4",
+    "@angular-eslint/builder": "~14.0.0",
+    "@angular-eslint/eslint-plugin": "~14.0.0",
+    "@angular-eslint/eslint-plugin-template": "~14.0.0",
+    "@angular-eslint/schematics": "~14.0.0",
+    "@angular-eslint/template-parser": "~14.0.0",
+    "@angular/cli": "^14.0.4",
+    "@angular/compiler-cli": "^14.0.4",
+    "@angular/language-service": "^14.0.4",
+    "@cypress/schematic": "^2.0.0",
     "@floogulinc/cypress-mongo-seeder": "^1.1.1",
-    "@types/jasmine": "^3.8.2",
-    "@types/node": "^16.11.26",
-    "@typescript-eslint/eslint-plugin": "^5.25.0",
-    "@typescript-eslint/parser": "^5.27.1",
-    "cypress": "^9.7.0",
-    "eslint": "^8.17.0",
+    "@types/jasmine": "^4.0.3",
+    "@types/node": "^18.0.0",
```

Comment:
Do we want to update the Node types to greater than the current LTS release we are using everywhere (16)?

Reply:
I don't know. Is the types system smart enough to know which version of Node that we're using and actually apply that set of types? I'm not sure how backwards compatible these tools are.

We presumably want to switch to Node v18 sometime anyway, so if this isn't breaking things we could just leave it.

Reply:
It's always going to be safer to use types from older versions. Additionally, 18 is not a LTS release so we won't be switching to it. 16 is the current LTS release until 20 releases in April 2023.

Reply:
Thanks for the info about 18 not being LTS. I think I somehow assumed all the even numbered versions were LTS. Oops.

Reply:
> Thanks for the info about 18 not being LTS. I think I somehow assumed all the even numbered versions were LTS.

Oh, TIL, thanks! 🙂

Reply:
So I've done some reading and I _think_ that we should be safe upgrading `@types/node` (or any of the `@types/*` dependencies) to higher versions than the thing that we're actually using. The typing system supports/expects information on when a feature was introduced/modified/etc., and seems to do the typing based on the major semver number. (I found [this](https://stackoverflow.com/questions/42035263/relationship-between-the-version-of-node-js-and-the-version-of-types-node) and [that](https://github.com/definitelytyped/definitelytyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library) useful, as well as just looking at some of the `*.d.ts` files [in `types/node`](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node).)

So I think we should be fine having a newer version of a `@types` file, as it should (dangerous words, I realize) be able to recognize the versions we're actually using and apply the correct typing.

Dependabot "wants" us to upgrade `@types/node` (see #1037), so actually doing so will avoid an ongoing wrestling match with that, which would be nice.

So unless we have some evidence that the typing system doesn't use versions the way I think they are (or that that whole system is busted in some way), I'd be inclined to go ahead and let this version change happen.

Thoughts? (esp. @floogulinc)

Reply:
Typescript has no way to "detect" which version of node you're actually using. The types packages are aligned to the versions of the packages they are types for. Eh @types/node@18.x.x is types for Node.js 18 and @types/node@16.x.x is types for Node.js 16. There is no mechanism for typescript to know to use anything else. GitHub actually recently fixed dependabot to match your types with the actual version of the package you use: https://github.blog/changelog/2022-05-10-dependabot-keeps-types-dependencies-in-sync-with-updated-packages/
This doesn't work for Node itself though because Node itself isn't a npm package.

The problem with using newer types than the version of node we use for builds and testing is that users may attempt to use a feature only added in newer versions of Node and typescript will be happy because the types match what you've declared as the node version for the types.

Reply:
Additionally, https://github.com/definitelytyped/definitelytyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library is talking about their internal publishing system and how it determines which version number to publish from a given set of types.

Reply:
It seems GitHub is currently recommending you have dependabot ignore the @types/node package https://github.com/github-community/community/discussions/16339#discussioncomment-2917404

Reply:
You've convinced me 😜. I've rolled back `@types/node` to v16 and added the bit telling Dependabot to ignore `@types/node` updates. That does mean that we'll have to remember to move that forward "by hand" when we do move to a new version of node, and that's exactly the kind of thing that I tend to forget, but we'll live.